src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -0,0 +1,1101 @@
+        string? runSettings,
+        TestPlatformOptions? options,
+        TestSessionInfo? testSessionInfo,
+        ITestRunEventsHandler testRunEventsHandler)
+    {
+        var sourceList = sources.ToList();
+        _testPlatformEventSource.TranslationLayerExecutionStart(
+            0,
+            0,
+            sourceList.Count,
+            runSettings ?? string.Empty);
+
+        try
+        {
+            TestRequestManager?.ResetOptions();
+
+            var testRunPayload = new TestRunRequestPayload
+            {
+                Sources = sourceList.ToList(),
+                RunSettings = runSettings,
+                TestPlatformOptions = options,
+                TestSessionInfo = testSessionInfo
+            };
+
+            TestRequestManager?.RunTests(
+                testRunPayload,
+                customTestHostLauncher: null,
+                new RunHandlerToEventsRegistrarAdapter(testRunEventsHandler),
+                new ProtocolConfig { Version = _highestSupportedVersion });
+        }
+        catch (Exception ex)
+        {
+            EqtTrace.Error("InProcessVsTestConsoleWrapper.RunTests: Exception occurred: " + ex);
+            var testRunCompleteArgs = new TestRunCompleteEventArgs(null, false, true, ex, null, null, TimeSpan.MinValue);
+
+            testRunEventsHandler.HandleLogMessage(TestMessageLevel.Error, ex.ToString());
+            testRunEventsHandler.HandleTestRunComplete(testRunCompleteArgs, null, null, null);
+        }
+
+        _testPlatformEventSource.TranslationLayerExecutionStop();
+    }
+
+    /// <inheritdoc/>
+    public void RunTests(
+        IEnumerable<TestCase> testCases,
+        string? runSettings,
+        ITestRunEventsHandler testRunEventsHandler)
+    {
+        RunTests(
+            testCases,
+            runSettings,
+            options: null,
+            testRunEventsHandler);
+    }
+
+    /// <inheritdoc/>
+    public void RunTests(
+        IEnumerable<TestCase> testCases,
+        string? runSettings,
+        TestPlatformOptions? options,
+        ITestRunEventsHandler testRunEventsHandler)
+    {
+        RunTests(
+            testCases,
+            runSettings,
+            options,
+            testSessionInfo: null,
+            testRunEventsHandler);
+    }
+
+    /// <inheritdoc/>
+    public void RunTests(
+        IEnumerable<TestCase> testCases,
+        string? runSettings,
+        TestPlatformOptions? options,
+        TestSessionInfo? testSessionInfo,
+        ITestRunEventsHandler testRunEventsHandler)
+    {
+        var testCaseList = testCases.ToList();
+        _testPlatformEventSource.TranslationLayerExecutionStart(
+            0,
+            0,
+            testCaseList.Count,
+            runSettings ?? string.Empty);
+
+        try
+        {
+            TestRequestManager?.ResetOptions();
+
+            var testRunPayload = new TestRunRequestPayload
+            {
+                TestCases = testCases.ToList(),
```

Comment:
Do we need the `ToList` call here? Can't we use `testCaseList` instead?

Reply:
Fixed.

Reply:
Not fixed